lib/train/transports/ssh_connection.rb

```diff
@@ -350,21 +353,30 @@ def execute_on_channel(cmd, opts)
           end
         end
       end
+      session.loop
 
-      thr = Thread.new { session.loop }
-
-      if timeout
-        res = thr.join(timeout)
-        unless res
-          logger.debug("train ssh command '#{cmd}' reached requested timeout (#{timeout}s)")
-          session.channels.each_value { |c| c.eof!; c.close }
-          raise Train::CommandTimeoutReached.new "ssh command reached timeout (#{timeout}s)"
-        end
-      else
-        thr.join
+      if timeout && timeoutable?(cmd) && exit_status == 124
```

Comment:
Just a thought, does maintaining constant values for exit status in this case makes sense? So instead of 124 if it said something like `exit_status == SIGTERM_EXIT_CODE` and then maintaining values of these exit codes in a common file maybe called as `constants.rb`? This way we can avoid hardcoding values in our code and attach meaning to it.